sdk/ml/azure-ai-ml/azure/ai/ml/_arm_deployments/arm_templates/workspace_base.json

```diff
@@ -807,7 +806,54 @@
                     "resourceId": "[parameters('byo_open_ai_resource_id')]"
                 },
                 "serverlessComputeSettings": "[parameters('serverless_compute_settings')]"
-            }
+            },
+            "resources":[
+                {
+                    "condition": "[equals(parameters('set_up_feature_store'), 'true')]",
+                    "type": "connections",
+                    "apiVersion": "2022-05-01",
+                    "name": "[variables('offlineStoreConnectionName')]",
+                    "location": "[parameters('location')]",
+                    "dependsOn": [
+                        "[resourceId('Microsoft.MachineLearningServices/workspaces', parameters('workspaceName'))]",
+                        "[variables('offlineStoreConnectionTarget')]"
```

Comment:
what is the difference between using dependsOn the variable, vs using condition, like in the previous way?
 "condition": "[and(equals(parameters('set_up_feature_store'), 'true'), not(empty(variables('offlineStoreConnectionName'))))]",

Reply:
dependsOn controls the order of step's execution. condition controls whether the step should be executed.
We don't need not(empty(variables('offlineStoreConnectionName'))))] in condition, but only need it to be executed after offline store target is there, that's because we will always have that offline store target (either provisioned by us or user provided)